x-pack/plugins/observability_solution/apm/public/components/app/settings/custom_link/create_edit_custom_link_flyout/filters_section.tsx

```diff
@@ -29,9 +29,9 @@ export function FiltersSection({
   onChangeFilters: (filters: Filter[]) => void;
 }) {
   const onChangeFilter = (key: Filter['key'], value: Filter['value'], idx: number) => {
-    const newFilters = [...filters];
-    newFilters[idx] = { key, value };
-    onChangeFilters(newFilters);
+    const updatedFilters = filters.map((filter) => ({ ...filter, isInitializing: false }));
```

Comment:
To avoid repeating this and optimize it a bit we can have it before the function using `useMemo`
```js
  const updatedFilters = useMemo(
    () => filters.map((filter) => ({ ...filter, isInitializing: false })),
    [filters]
  );
```
and remove it from the functions